src/Controller/ExerciseController.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Exercise;
+use App\Form\ExerciseType;
+use App\Repository\ExerciseRepository;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/exercise")
+ */
+class ExerciseController extends BaseController
+{
+    /**
+     * @Route("/", name="exercise_index", methods={"GET"})
+     */
+    public function index(ExerciseRepository $exerciseRepository): Response
+    {
+        return $this->render('exercise/index.html.twig', [
+            'exercises' => $exerciseRepository->findAll(),
+        ]);
+    }
+
+    /**
+     * @Route("/new", name="exercise_new", methods={"GET","POST"})
+     */
+    public function new(Request $request): Response
+    {
+        $exercise = new Exercise();
+        $form = $this->createForm(ExerciseType::class, $exercise);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            if ($form->get('intervals')->getData() == null) {
+                $exercise->setIntervals(1);
+            }
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($exercise);
+            $entityManager->flush();
+
+            $this->addSuccessFlash(sprintf('Successfully created exercise "%s"', $exercise->getName()));
+            return $this->redirectToRoute('exercise_index');
+        }
+
+        return $this->render('exercise/new.html.twig', [
+            'exercise' => $exercise,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{id}", name="exercise_show", methods={"GET"})
+     */
+    public function show(Exercise $exercise): Response
+    {
+        return $this->render('exercise/show.html.twig', [
```

Comment:
@daria-sieroshtan 
what you are doing is fine but I would like to remind you that you can use `@Template` annotation for clarity

Reply:
I do remember.
This part is auto-generated so I left it as is.